pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -175,6 +175,22 @@
         }
     }
 
+    protected CompletableFuture<List<String>> internalGetListAsync() {
+        return validateNamespaceOperationAsync(namespaceName, NamespaceOperation.GET_TOPICS)
+                .thenCompose(__ -> {
+                    try {
+                        if (!namespaceResources().namespaceExists(namespaceName)) {
+                            throw new RestException(Status.NOT_FOUND, "Namespace does not exist");
+                        }
+                    } catch (MetadataStoreException e) {
```

Comment:
If change ``namespaceResources().namespaceExists(namespaceName)`` to async, we can remove this catch.